### ISSUE TITLE
Error while running BRAKER3

### ISSUE THREAD

**Issue by InesGaLu** [Gaius-Augustus/BRAKER#880]
Hello everyone!

I am trying to annotate a genome using BRAKER3. I have already done the previous steps (like mapping the RNA-seq to the genome using STAR), but now I'm encountering several problems while running BRAKER3. I feel like I'm at a dead-end. 

The error files is like this: 
``ERROR in file /users-d2/ines.galan/programmes/BRAKER-3.0.8/scripts/braker.pl at line 5575
Failed to execute: /anaconda3/bin/perl /programmes/GeneMark-ETP/bin/gmetp.pl --cfg /gd/braker_single_input/braker/GeneMark-ETP/etp_config.yaml --workdir /gd/braker_single_input/braker/GeneMark-ETP --bam /gd/braker_single_input/braker/GeneMark-ETP/etp_data/ --cores 16 --softmask  1>/gd/braker_single_input/braker/errors/GeneMark-ETP.stdout 2>/gd/braker_single_input/braker/errors/GeneMark-ETP.stderr
Failed to execute: /users-d2/ines.galan/anaconda3/bin/perl /users-d2/ines.galan/programmes/GeneMark-ETP/bin/gmetp.pl --cfg /gd/braker_single_input/braker/GeneMark-ETP/etp_config.yaml --workdir /gd/braker_single_input/braker/GeneMark-ETP --bam /gd/braker_single_input/braker/GeneMark-ETP/etp_data/ --cores 16 --softmask  1>/gd/braker_single_input/braker/errors/GeneMark-ETP.stdout 2>/gd/braker_single_input/braker/errors/GeneMark-ETP.stderr
The most common problem is that GeneMark-ETP didn't receive enough evidence from the input data, in this case, see > errors/GeneMark-ETP.stderr!

Then, if I go and check those specific errors, I get: 
~/gd/braker_single_input/braker/errors$ cat GeneMark-ETP.stderr
Use of uninitialized value $_ in concatenation (.) or string at /users/programmes/GeneMark-ETP/bin/gms2hints.pl line 1013, <$IN> line 9.
error, unexpected line format found:  at /users/programmes/GeneMark-ETP/bin/gms2hints.pl line 1013, <$IN> line 9.
error, file/folder not found: genome_gmst.gtf

But it doesn't find it because the program itself should create it. I don't know how to continue and I would appretiate any posible help!

**Comment by Carrefour-pirate** [comment]
mark

**Comment by Carrefour-pirate** [comment]
I just meet the same trouble. With the exactly same data once succeed.

**Comment by InesGaLu** [comment]
Did you manage to make it work?

**Comment by Carrefour-pirate** [comment]
> Did you manage to make it work?
not yet, trying singularity now.

**Comment by Carrefour-pirate** [comment]
not yet, trying singularity now.

**Comment by InesGaLu** [comment]
Good morning! Are you still dealing with this problem? I'm stuck at the same point and I am guessing I'll have to work with singularity

**Comment by Carrefour-pirate** [comment]
sorry for the delay. 
and i  gave up with braker3 for this trouble. 
i failed deploying braker3 singularity mirror because of net connection, and i don't  know whether chinese great fire wall or the braker3 self caused this.
i finally finished my job with braker2.
you can still try singularity if you are not at china, it's easy to get a tutorial.
or, braker2 may be an alternative option if no other option.

**Comment by ice-cold** [comment]
Hi everyone,

I encountered the same error and after some testing, I found that it seems to be caused by outdated packages, such as gffread and StringTie. Even though I’ve installed Cufflinks and StringTie1 on my machine a long time ago, and added the /GeneMark-ETP/tools directory (which contains the necessary dependencies) to my $PATH, the old packages were still being prioritized by BRAKER3.

After removing these old packages, BRAKER3 was able to correctly access the tools in /GeneMark-ETP/tools and successfully completed the test3.sh script in the example folder, resolving the error.

So, make sure BRAKER3 is using the correct version of the dependency tools. Hope this helps!

yec

**Comment by Carrefour-pirate** [comment]
yes, it works. i softlink all files in GeneMark-ETP/tools to envs/braker3/bin, and rm all same files out $PATH, then this trouble disappears.
thanks bro, or maybe sis.

**Comment by KatharinaHoff** [comment]
The monolithic braker.pl Perl script has been replaced entirely in [BRAKER4](https://github.com/Gaius-Augustus/BRAKER4). The pipeline is now a Snakemake workflow where each step is a separate rule running inside a Singularity container. All file path management, directory creation, and tool invocation are handled by Snakemake. This class of braker.pl errors no longer exists. Please try BRAKER4 and report any issues there.
